### PR TITLE
Add AWS C++ SDK to provision scripts

### PR DESCRIPTION
**Pull request by zwass**

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: tools/provision/darwin.sh [hunk 1]
```diff
@@ -36,4 +36,5 @@ function main_darwin() {
   package google-benchmark
   package libmagic
   package sleuthkit
+  package aws-sdk-cpp
```
Comment (zwass) [inline tools/provision/darwin.sh]: Heads up that this won't work until https://github.com/Homebrew/homebrew-core/pull/109 is accepted. Until then, on Mac one can do `brew install https://raw.githubusercontent.com/zwass/homebrew-core/aws-sdk-cpp/Formula/aws-sdk-cpp.rb --with-static --without-http-client`.

**Comment by zwass** [comment]
Homebrew has merged the formula pull request, so this should be good to test. Once this has been merged, we can test https://github.com/facebook/osquery/pull/2045.

**Comment by marpaia** [comment]
ok to test

**Comment by osqueryer** [comment]
Build origin/pr/2046/merge (Job: 2611) is good! :+1:

**Comment by zwass** [comment]
AFAICT the Jenkins build does not run `make deps`, which would make the Jenkins results we just saw effectively a noop.

**Comment by marpaia** [comment]
Just talking to @javuto, we'll have to re-provision the build slaves to get these changes to take effect. Javier is currently out, but he'll be back on Monday to assist.

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
Build origin/pr/2046/merge (Job: 2614) is good! :+1:

**Pull request merged**